File: Programmers/1/118666. 성격 유형 검사하기/성격 유형 검사하기.java
import java.util.*;

class Solution {
    public String solution(String[] survey, int[] choices) {
        String answer = "";
        Map<Character, Integer> scoreMap = new HashMap<>();

        for (char type : new char[]{'R', 'T', 'C', 'F', 'J', 'M', 'A', 'N'}) {
            scoreMap.put(type, 0);
        }

        for (int i = 0; i < survey.length; i++) {
            char disagreeType = survey[i].charAt(0);
            char agreeType = survey[i].charAt(1);
            int choice = choices[i];

            if (choice < 4) {
                scoreMap.put(disagreeType, scoreMap.get(disagreeType) + (4 - choice));
            } else if (choice > 4) {
                scoreMap.put(agreeType, scoreMap.get(agreeType) + (choice - 4));
            }
        }

        StringBuilder result = new StringBuilder();
        result.append(scoreMap.get('R') >= scoreMap.get('T') ? 'R' : 'T');
        result.append(scoreMap.get('C') >= scoreMap.get('F') ? 'C' : 'F');
        result.append(scoreMap.get('J') >= scoreMap.get('M') ? 'J' : 'M');
        result.append(scoreMap.get('A') >= scoreMap.get('N') ? 'A' : 'N');

        answer = result.toString();
        
        return answer;
    }
}

File: Programmers/1/118666. 성격 유형 검사하기/성격 유형 검사하기.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

string solution(vector<string> survey, vector<int> choices) {
    string answer = "";
    unordered_map<char, int> scoreMap;

    string types = "RTCFJMAN";
    
    for (char c : types) {
        scoreMap[c] = 0;
    }

    for (int i = 0; i < survey.size(); i++) {
        char disagree = survey[i][0];
        char agree = survey[i][1];
        int choice = choices[i];

        if (choice < 4) {
            scoreMap[disagree] += 4 - choice;
        } else if (choice > 4) {
            scoreMap[agree] += choice - 4;
        }
    }

    answer += (scoreMap['R'] >= scoreMap['T']) ? 'R' : 'T';
    answer += (scoreMap['C'] >= scoreMap['F']) ? 'C' : 'F';
    answer += (scoreMap['J'] >= scoreMap['M']) ? 'J' : 'M';
    answer += (scoreMap['A'] >= scoreMap['N']) ? 'A' : 'N';
    
    return answer;
}